nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/ExecuteScriptMarkLogic.java

```diff
@@ -298,10 +307,22 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
                 resultToFlowFile(session, last, lastFF, resultsDest);
                 session.transfer(lastFF, LAST_RESULT);
             }
-
+            session.commitAsync();
+        } catch (com.marklogic.client.FailedRequestException ex) {
+            // If there's an error running the script, rolling back the session and trying again isn't likely to work.
+            // Instead, transfer the flowfile to ERROR so that the flow can handle the error. 
+            getLogger().info("Error in ExecuteScriptMarkLogic. Sending to ERROR");
+            logError(ex);
+            originalFF = session.putAttribute(originalFF, MARKLOGIC_RESULTS_ERROR, ex.getMessage());
+            session.transfer(originalFF, ERROR);
             session.commitAsync();
         } catch (final Throwable t) {
-            this.logErrorAndRollbackSession(t, session);
+            // Other errors could mean that MarkLogic is unreachable or some other problem. Transfer to FAILURE. 
+            logError(t);
+            getLogger().info("Error Rolling back session " );
```

Comment:
For better or worse, I am thinking we should continue to use `logErrorAndRollbackSession` for consistency with how the other processors work when they catch a `Throwable`. I noticed that the javadocs for that method caution: "This may not be desirable to a user; it may be better to route the FlowFile to a failure relationship". The ability to route a general failure to a relationship (instead of rolling the session back and rethrowing a ProcessException) seems like a feature to be implemented on all the processors currently making use of this method. 

That would mean the scope of this issue is - is we get a `FailedRequestException`, we can safely conclude that rolling the session back is not going to help anything, and we should always route this to a 'failed request' relationship. Otherwise, we'll keep doing what we've been doing - again, for better or worse.

Reply:
I'm changing my mind on this now - I like what `ExtensionCallML` does, which is to store the error message as an attribute and route it to "Failure". That seems like the right behavior in all cases, really. But at least for `ExtensionCallML` and `ExecuteScriptML` and now `CallRestExtensionML`, I think we should do this behavior.